Feat: AI Azure Content Safety plugin doc (#7326)
Kong/docs.konghq.com: app/_hub/kong-inc/ai-azure-content-safety/how-to/_index.md

### Managed Identity Auth

To use Managed Identity auth (e.g. Machine Identity on an Azure VM or AKS Pod), you must set `config.use_azure_managed_identity`
to `true`.

Following this, there are three more parameters that may or may not be required:

* `config.azure_client_id`
* `config.azure_client_secret`
* `config.azure_tenant_id`

The client ID is normally required when you want to use a different *user assigned identity* instead of the 
managed identity assigned to the resource on which Kong is running.

The client secret and tenant ID are usually only used when you are running Kong somewhere outside 
of Azure, but still want to use Entra ID (ADFS) to authenticate with Content Services.

See the [cloud provider authentication](/hub/kong-inc/ai-proxy/how-to/cloud-provider-authentication/) guide to learn more.

## Examples

Configure the plugin with an array of supported categories as defined in the 
[Content Services REST API documentation](https://westus.dev.cognitive.microsoft.com/docs/services/content-safety-service-2023-10-01/operations/TextOperations_AnalyzeText), for example using all four

> supported categories in this API version:

Author: It might actually be more useful to link to https://learn.microsoft.com/en-us/azure/ai-services/content-safety/concepts/harm-categories?tabs=warning (or maybe both?).

> Configure the plugin with an array of supported categories as defined in the 
> [Content Services REST API documentation](https://westus.dev.cognitive.microsoft.com/docs/services/content-safety-service-2023-10-01/operations/TextOperations_AnalyzeText), for example using all four
> supported categories in this API version:

Author: ```suggestion
Configure the plugin with an array of supported categories, as defined by Azure Content Safety:
* [Content Services REST API documentation](https://westus.dev.cognitive.microsoft.com/docs/services/content-safety-service-2023-10-01/operations/TextOperations_AnalyzeText)
* [Harm categories in Azure AI Content Safety](https://learn.microsoft.com/en-us/azure/ai-services/content-safety/concepts/harm-categories)

Azure's harm categories map to `categories.name` in the plugin's configuration, and the severity levels map to `categories.rejection_level`. 

For example, here's what it looks like if you use all four
supported categories in this API version:
```

<!-- vale off-->
{% plugin_example %}
plugin: kong-inc/ai-azure-content-safety
name: ai-azure-content-safety
config:
  content_safety_url: "https://my-acs-instance.cognitiveservices.azure.com/contentsafety/text:analyze"
  use_azure_managed_identity: false
  content_safety_key: "{vault://env/AZURE_CONTENT_SAFETY_KEY}"
  categories:
    - name: Hate

>       rejection_level: 2

Author: our `rejection_level` maps to `severity`.

    - name: SelfHarm
      rejection_level: 2
    - name: Sexual
      rejection_level: 2
    - name: Violence
      rejection_level: 2
  text_source: concatenate_user_content
  reveal_failure_reason: true
  output_type: FourSeverityLevels  # Supports FOUR or EIGHT level severity-grading
targets:
  - service
formats:
  - konnect
  - curl
  - yaml
  - kubernetes
{% endplugin_example %}
<!--vale on -->

Now, given the following AI Chat request:

```json
{
  "messages": [
    {
      "role": "system",
      "content": "You are a mathematician."
    },
    {
      "role": "user",
      "content": "What is 1 + 1?"
    },
    {
      "role": "assistant",
      "content": "The answer is 3."
    },
    {
      "role": "user",
      "content": "You lied, I hate you!"
    }
  ]
}
```

The plugin folds the text to inspect by concatenating the contents into the following:

```plaintext
You are a mathematician.; What is 1 + 1?; The answer is 3.; You lied, I hate you!
```

Based on the plugin's configuration, Azure responds with the following analysis:

```json
{
    "categoriesAnalysis": [
        {
            "category": "Hate",
            "severity": 2
        }
    ]
}
```

> This breaches the plugin's configured (inclusive and greater) threshold of `2` for `Hate` and `Violence`, and sends a 400 error code to the client:

Reviewer: And how does it breach violence as well as hate?

> This breaches the plugin's configured (inclusive and greater) threshold of `2` for `Hate` and `Violence`, and sends a 400 error code to the client:

Author: It is explained in the Azure docs, not really sure how this breaches violence though either.

> This breaches the plugin's configured (inclusive and greater) threshold of `2` for `Hate` and `Violence`, and sends a 400 error code to the client:

Author: ```suggestion
This breaches the plugin's configured (inclusive and greater) threshold of `2` for `Hate`, and sends a 400 error code to the client:
```

```json
{
	"error": {

> 		"message": "request failed content safety check: breached category [Hate] at level 2; breached category [Violence] at level 2"

Author: ```suggestion
		"message": "request failed content safety check: breached category [Hate] at level 2"
```

	}
}
```

### Hiding the failure from the client

If you don't want to reveal to the caller why their request has failed, you can set `config.reveal_failure_reason` to `false`, in which
case the response looks like this:

```json
{
	"error": {
		"message": "request failed content safety check"
	}
}
```

> ### Using Blocklists

Author: ```suggestion
### Using blocklists
```

The plugin supports previously-created blocklists in Azure Content Safety.

Using the [Azure Content Safety API](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/contentsafety/operation-groups) 
or the Azure Portal, you can create a series of blocklists for banned phrases or patterns. 

> You can then reference their unique names in the plugin configuration. For example:

Reviewer: Could we provide a small description of what this example is blocking?

> You can then reference their unique names in the plugin configuration. For example:

Author: ```suggestion
You can then reference their unique names in the plugin configuration. 

In the following example, the plugin takes two existing blocklists from Azure, `company_competitors` and 
`financial_properties`:
